ion-hash/src/lib.rs

```diff
@@ -106,21 +110,6 @@ where
 
     fn hash_scalar<E: Element + ?Sized>(&mut self, elem: &E) {
         self.hash_no_repr(elem);
-        if let Some(repr) = representation::binary_repr(elem) {
-            self.hasher.update(ion_hash_escape(&repr[..]));
-        }
+        representation::write_repr(elem, &mut self.hasher);
```

Comment:
I got confused by this terminology

`no_repr` is confusing, I might spell this as `hash_type` or something, because IIRC, you need to do this for all values, right?  `write_repr` might be better spelled as `scalar_repr` or something like that.

Channeling my inner @zslayton also I would suggest we spell out the terms to be consistent with the style of the other crates (e.g. `representation`).

Reply:
Ok!